inst/extdata/sharedcache/demo.R

```diff
@@ -1,116 +0,0 @@
-# timestamp helpers for cache/indicator/data functions to follow
-
-#' formats the modified-time of a file into a character string
-record_mtime <- function(filename) {
-  POSIX2char(file.mtime(filename))
-}
-#' reads a timestamp from an .ind file into POSIXct
-read_time <- function(indfile) {
-  char2POSIX(readLines(indfile))
-}
-
-#' Create length-3 vector of cache, indicator, and data filenames
-#' 
-#' For shared cache: given any name ending in .cache, .ind, or something else,
-#' categorizes the name and finds matching versions of .cache, .ind, and data
-#' filenames
-expand_names <- function(filename) {
-  # the result will be a named vector
-  allnames <- list()
-  
-  # determine the name of the datafile
-  if(grepl('\\.cache$', filename)) allnames$data <- gsub('.cache', '', filename)
-  else if(grepl('\\.ind$', filename)) allnames$data <- gsub('.ind', '', filename)
-  else allnames$data <- filename
-  
-  allnames$cache <- paste0(allnames$data, '.cache')
-  allnames$indicator <- paste0(allnames$data, '.ind')
-  allnames$var <- tools::file_path_sans_ext(allnames$data)
-  
-  return(allnames)
-}
-
-
-# Mock functions for checking, pulling, and pushing to cache, and processing a
-# file
-
-#' get a hash of the cache file. for an S3 cache, this code could be replaced by
-#' S3's ETag (an MD5 sum, with special treatment for multi-part uploads,
-#' supplied in list_s3)
-get_hash <- function(fname) {
-  unname(tools::md5sum(fname))
-}
-
-# make_file must be defined in both helper-demo.R and extdata/sharedcache/demo.R
-# because we need it both places, remake only sees one
-make_file <- function(fname, ftext='', ftstamp=Sys.time()) { # "2017-09-05 07:00:00 MST"
-  readr::write_lines(ftext, fname)
-  system(sprintf('touch -d "%s" %s', POSIX2char(ftstamp), fname))
-  invisible(NULL)
-}
-
-make_cache_indicator <- function(indfile) {
-  fnames <- expand_names(indfile)
-  if(!file.exists(fnames$cache)) stop(paste(fnames$data, "is missing from the cache"))
-  message("note ", fnames$var)
-  
-  # write the indicator file
-  sc_indicate(fnames$indicator, data_file=fnames$cache)
-  invisible(NULL)
-}
```

Comment:
there's a bunch of outdated proof-of-concept stuff in here. scipiper does a lot of this differently now.

Reply:
Good to know. I started playing around in here to find an example outside of the vignette and was scratching my head...

Reply:
Yeah, the old tests weren't so last year :stuck_out_tongue_closed_eyes: . scipiper does have a fair amount of tech debt - we're doing a lot of testing and repairing and augmenting on the fly, in conjunction with application development, but the lack of vignettes and formal tests may catch up with us, as could the fact that there is no Travis connection and even local automated Checks and Tests fail miserably right now.

...that said, I think we should continue to balance investment in scipiper with the understanding that our tools could change beneath our feet in not too long (cough cough drake), which is why I waited until today to work on these tests.